### DISCUSSION TITLE
Überwachung der Zelltemperatur mehrerer Pylontech-Batterien und anderer wichtiger Funktionen über RS485

### DISCUSSION THREAD

**Discussion by Manos1966** [skippermeister/OpenDTU-OnBattery#2] [Show and tell]
Diese OpenDTU Version kann vieles machen.
Eine Besonderheit ist das Ablesen wichtiger Informationen aus einer, oder mehrerer Pylontech Batterien ueber die RS485 Schnittstelle:

![01 Slave-1](https://github.com/skippermeister/OpenDTU-OnBattery/assets/129660360/911b3f59-a1cd-431f-a491-1dbf2fd4cb2a)

Zweite Besonderheit, die CANbus Schnittstelle kann gleichzeitig von anderen Anwendungen (z.B. zweite OpenDTU) benutzt werden.

Zwei verschiedene RS485 Platinen werden unterstützt (waehlen Sie entsprechend das "Ausgewaelte Profil" in der Hardwae-Einstellungen::
"Type 1"-Adapter
![Type 1 RS485 Modul - Copy](https://github.com/skippermeister/OpenDTU-OnBattery/assets/129660360/5fad9433-d447-42f6-bc98-124dd7b6b75a)

"Type 2"-Adapter
![Type 2 TTL zu RS485](https://github.com/skippermeister/OpenDTU-OnBattery/assets/129660360/4feec812-e5fa-47eb-a5a7-34482983443c)

Beachte: das Breakoutboard kann mit einer RJ45 Buchse bestückt werden. Dann ist vom Breakoutboard Pin 4+5 auf Masse gelegt. In diesem Fall auf keinen Fall ein fertiges 1:1 Kabel verwenden.
Am einfachsten geht es jedoch indem Du einfach wie folgt vorgehst:
Nimm ein RJ45 Ethernet Kabel, schneide es auf die benötigte Länge durch und lege die Adern auf der abgeschnittene Seite frei.
Alternativ, einen Adapter nutzen:
![RJ45 solution](https://github.com/skippermeister/OpenDTU-OnBattery/assets/129660360/abc735c4-f6fb-4119-9f6b-d75fa2623e14)

Verbinde
**Breakoutboard A an Pylontech A/CAN RJ45 Pin 7 (das ist 485A)
Breakoutboard B an Pylontech A/CAN RJ45 Pin 8 (das ist 485B)**
**Die Pins 1 bis 6 auf der Pylontech Seite unbedingt offen lassen.**  Ansonsten kann es (wird es!) Störungen geben. RS485 ist ein Gegentaktsignal.
![Pin Layout RS485](https://github.com/skippermeister/OpenDTU-OnBattery/assets/129660360/7d55550c-d1ae-4e08-b066-7bfd3511ed14)

Das RS485-RJ45 Kabel wird an die MASTER-Pylontech Batterie angeschlossen.

**Comment by Manos1966** [comment]
Die Pylontech Batterien geben viele Informationen Preis ueber die RS485 Schnittstelle.
Die OpenDTU liefert diese Informationen eins zu eins weiter.
Manche stehen nicht im Einklang zu der Informationen der Bedienungsanleitung:

![image](https://github.com/skippermeister/OpenDTU-OnBattery/assets/146529906/ab661476-1226-49d2-9e15-e999d1c9b78f)
![Capture2](https://github.com/skippermeister/OpenDTU-OnBattery/assets/129660360/c2ad7abb-351e-4205-a88f-79d10b7cab45)

Die empfohlenen Temperaturbereiche, welche aus den System Parametern der Batterien ausgelesen werden, weichen deutlich ab.

**Comment by Manos1966** [comment]
Hallo @skippermeister 
ich habe ein paar Infos hier gepostet. 
Du kannst sie beliebig ändern, falls du Verbesserungen siehst 😉 👍

**Reply by Manos1966** [reply]
@skippermeister Uebrigens im Wiki, die Bilder sind genau umgekehrt.

. Pylontech Batterie connected via RS485 using two different RS485 Modules
Type1: a module with a simple MAX3485 or SP3485 Chip, needs RX, TX and RTS (/RE DE)
Die ist die ROTE Platine

Type2: a TTL to RS485 module with an automatic self driven /RE DE logic. Needs only RX and TX lines
Dies ist die BLAUE Platine

**Reply by skippermeister** [reply]
Danke für den Hinweis. Ich hab es korrigiert. Es war wohl zu spät in der Nacht :-)
